versioned/persist/nontx/src/main/java/org/projectnessie/versioned/persist/nontx/NonTransactionalDatabaseAdapter.java

```diff
@@ -208,8 +210,21 @@ public Hash merge(
                 writeGlobalCommit(
                     ctx, timeInMicros, Hash.of(pointer.getGlobalId()), Collections.emptyList());
 
+            Hash newRefLogId =
+                writeRefLogEntry(
```

Comment:
This and all other call sites of `writeRefLogEntry` look tike they will be duplicated in `TxDatabaseAdapter` - better to not have duplicated code.

Reply:
yeah. If I move this to abstract class, then I need to use `RefLog` instead of proto `ReflogEntry`.
So, it brings an extra conversion for each operation. 
Considering code duplication vs perf. I chose performance. 

Actually I tired adding proto dependency to abstract class. but its a circular dependency. I think some part is not designed common for txn and non-txn.